packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,204 @@
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listActive', request);
+        const { uid } = handleAuth(request.auth);
+        return db.fetchAccountSubscriptions(uid);
+      }
+    },
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/active',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken'
+        },
+        validate: {
+          payload: {
+            planId: validators.subscriptionsPlanId.required(),
+            paymentToken: validators.subscriptionsPaymentToken.required()
+          }
+        },
+        response: {
+          schema: isA.object().keys({
+            subscriptionId: validators.subscriptionsSubscriptionId.required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.createSubscription', request);
+
+        const { uid, email } = handleAuth(request.auth);
+
+        await customs.check(request, email, 'createSubscription');
+
+        const { planId, paymentToken } = request.payload;
+
+        // Find the selected plan and get its product ID
+        const plans = await subhub.listPlans();
+        const selectedPlan = plans.filter(p => p.plan_id === planId)[0];
+        if (! selectedPlan) {
+          throw error.unknownSubscriptionPlan(planId);
+        }
+        const productName = selectedPlan.product_id;
+
+        const paymentResult = await subhub.createSubscription(uid, paymentToken, planId);
+
+        const subscriptionId = paymentResult.sub_id;
+        await db.createAccountSubscription({
+          uid,
+          subscriptionId,
+          productName,
+          createdAt: Date.now()
+        });
+
+        const devices = await request.app.devices;
+        await push.notifyProfileUpdated(uid, devices);
+        log.notifyAttachedServices('profileDataChanged', request, { uid, email });
+
+        log.info('subscriptions.createSubscription.success', { uid, subscriptionId });
+
+        return { subscriptionId };
+      }
+    },
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/updatePayment',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken'
+        },
+        validate: {
+          payload: {
+            paymentToken: validators.subscriptionsPaymentToken.required()
+          }
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.updatePayment', request);
+
+        const { uid, email } = handleAuth(request.auth);
+        await customs.check(request, email, 'updatePayment');
+
+        const { paymentToken } = request.payload;
+
+        await subhub.updateCustomer(uid, paymentToken);
+
+        log.info('subscriptions.updatePayment.success', { uid });
+
+        return {};
+      }
+    },
+    {
+      method: 'DELETE',
+      path: '/oauth/subscriptions/active/{subscriptionId}',
```

Comment:
Will all subscription cancellations occur on pages FxA controls or can the RP cancel a subscription?

Reply:
All subscription cancellations occur on FxA pages. RPs never have access to manage subscriptions.